src/lib.rs

```diff
@@ -259,15 +259,31 @@ fn serve_request_stream<Ctx: Clone>(
 ) -> impl Stream<Item = Result<Message, warp::Error>> {
     let resp_stream = srv
         .serve_ws(ctx, payload, service_id)
-        .map(move |payload_result| match payload_result {
-            Ok(payload) => Outgoing::Next {
-                request_id: req_id,
-                payload,
-            },
-            Err(kind) => Outgoing::Error {
-                request_id: req_id,
-                kind,
-            },
+        .take_until_condition(|resp| future::ready(resp.is_err()))
+        .ready_chunks(128)
+        .flat_map(move |payload_results| {
+            let mut err = None;
+            let mut payload = Vec::with_capacity(payload_results.len());
+            for payload_result in payload_results {
+                match payload_result {
+                    Ok(value) => payload.push(value),
+                    Err(kind) => err = Some(kind), // always comes last
+                }
+            }
+            let mut res = Vec::with_capacity(1);
```

Comment:
why not 2?

Reply:
i'd consider the success case more common.

Reply:
I’d use `Vec::new()` because it’s unlikely to matter, or `Vec::with_capacity(2)` as a hint to the reader, or `Vec::with_capacity(1)` if one element is unreasonably large.